pkg/component/kubeapiserverexposure/kube_apiserver_sni.go

```diff
@@ -144,7 +145,11 @@ func (s *sni) Deploy(ctx context.Context) error {
 		}); err != nil {
 			return err
 		}
-		if err := s.applier.ApplyManifest(ctx, kubernetes.NewManifestReader(envoyFilterSpec.Bytes()), kubernetes.DefaultMergeFuncs); err != nil {
+
+		filename := fmt.Sprintf("envoyfilter__%s__%s.yaml", envoyFilter.Namespace, s.namespace)
```

Comment:
Nit since I find it a bit less confusing:
```suggestion
		filename := fmt.Sprintf("envoyfilter__%s__%s.yaml", envoyFilter.Namespace, envoyFilter.Name)
```

Reply:
Included in https://github.com/gardener/gardener/pull/9005/commits/ebe0798b278ea2d255b4fd244e1a6305e8fb8091.